patchwork-fml/src/main/java/net/minecraftforge/fml/common/ObfuscationReflectionHelper.java

```diff
@@ -0,0 +1,340 @@
+ * the compatibility.
+ *
+ * <p>In other cases, AccessTransformers may be used.</p>
+ *
+ * <p>All field and method names should be passed in as intermediary names, and this will automatically resolve if Yarn mappings are detected.</p>
+ *
+ */
+@SuppressWarnings({ "serial", "unchecked", "unused", "WeakerAccess" })
+public class ObfuscationReflectionHelper {
+	private static final Logger LOGGER = LogManager.getLogger();
+	private static final Marker REFLECTION = MarkerManager.getMarker("REFLECTION");
+
+	/**
+	 * Remaps a class name using {@link PatchworkMappingResolver#remapName(INameMappingService.Domain, Class, String)}.
+	 * Throws an exception for non-class domains.
+	 *
+	 * @param domain The {@link INameMappingService.Domain} to use to remap the name.
+	 * @param name   The name to try and remap.
+	 * @return The remapped name, or the original name if it couldn't be remapped.
+	 * @throws UnsupportedOperationException if the {@code domain} is not {@link INameMappingService.Domain#CLASS}
+	 * @deprecated Patchwork: use {@link PatchworkMappingResolver} instead
+	 */
+	@Deprecated
+	@Nonnull
+	public static String remapName(INameMappingService.Domain domain, String name) {
+		if (domain == INameMappingService.Domain.CLASS) {
+			return PatchworkMappingResolver.remapName(domain, null, name);
+		} else {
+			// This would need a special remapping system, if someone uses it then it can be implemented
+			throw new UnsupportedOperationException("FIXME: Unable to lookup member of type " + domain.name() + " with name " + name + ".");
+		}
+	}
+
+	/**
+	 * Gets the value a field with the specified index in the given class.
+	 * Note: For performance, use {@link #findField(Class, int)} if you are getting the value more than once.
+	 *
+	 * <p>Throws an exception if the field is not found or the value of the field cannot be gotten.</p>
+	 *
+	 * @param classToAccess The class to find the field on.
+	 * @param instance      The instance of the {@code classToAccess}.
+	 * @param fieldIndex    The index of the field in the {@code classToAccess}.
+	 * @param <T>           The type of the value.
+	 * @param <E>           The type of the {@code classToAccess}.
+	 * @return The value of the field with the specified index in the {@code classToAccess}.
+	 * @throws UnableToAccessFieldException If there was a problem getting the field or the value.
+	 * @deprecated Use {@link #getPrivateValue(Class, Object, String)} because field indices change a lot more often than field names do.
+	 * Forge: TODO Remove in 1.15
+	 */
+	@Deprecated
+	@Nullable
+	public static <T, E> T getPrivateValue(Class<? super E> classToAccess, E instance, int fieldIndex) {
+		try {
+			return (T) findField(classToAccess, fieldIndex).get(instance);
+		} catch (Exception e) {
+			LOGGER.error(REFLECTION, "There was a problem getting field index {} from {}", fieldIndex, classToAccess.getName(), e);
```

Comment:
Does this actually print the exception?

Reply:
Yeah, it will print the exception and its stacktrace on a new line. I tested it